ast/ddl.go

```diff
@@ -1823,8 +1828,15 @@ func (n *TableOption) Restore(ctx *RestoreCtx) error {
 			ctx.WritePlain("''")
 		}
 	case TableOptionCharset:
-		ctx.WriteKeyWord("DEFAULT CHARACTER SET ")
-		ctx.WritePlain("= ")
+		if n.UintValue == TableOptionCharsetWithConvertTo {
+			ctx.WriteKeyWord("CONVERT TO ")
+		} else {
+			ctx.WriteKeyWord("DEFAULT ")
+		}
+		ctx.WriteKeyWord("CHARACTER SET ")
+		if n.UintValue == TableOptionCharsetWithoutConvertTo {
+			ctx.WriteKeyWord("= ")
+		}
```

Comment:
Perhaps make the equal sign unnecessary

Reply:
If we remove this equal sign, we should also remove the equal sign in `COLLATE = XXX`, and update a lot of tests.

Reply:
okay 😞

Reply:
I should say the syntax of MySQL is really weird.